sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala

```diff
@@ -257,6 +257,74 @@ class StreamingInnerJoinSuite extends StreamingJoinSuite {
     )
   }
 
+  test("stream stream inner join with one time-interval condition " +
+    "- with one watermark") {
+    val leftSource = MemoryStream[(Int, Int)]
+    val rightSource = MemoryStream[(Int, Int)]
+
+    val leftStream = leftSource
+      .toDF()
+      .withColumn("left_value", $"_1")
+      .withColumn("left_time", timestamp_seconds($"_2"))
+      .withWatermark("left_time", "15 seconds")
+
+    // Take note: no watermark is defined on the right
+    val rightStream = rightSource
+      .toDF()
+      .withColumn("right_value", $"_1")
+      .withColumn("right_time", timestamp_seconds($"_2"))
+
+    // Our condition is R > L + 10 and we have a watermark on the left, so state
+    // should be dropped on the right
+    val joined = leftStream
+      .join(
+        rightStream,
+        expr("left_value = right_value AND right_time > left_time + interval 10 seconds"),
+        "inner"
```

Comment:
is the fix only valid with one time-interval condition? what if adding right_time < left_time - interval 10 seconds?

Reply:
Yes, it's also valid, but you would need two watermarks in that case:

https://github.com/apache/spark/pull/44323/files#diff-a271c249e6989ebaad63a491910597cd336049516e5b8e9a80f2bbd7368f458dR478